### ISSUE TITLE
Logs flooded by Connecting to unix...

### ISSUE THREAD

**Issue by renanqts** [kubernetes-sigs/aws-efs-csi-driver#564]
/kind bug

**What happened?**
Logs are flooded by:
efs-csi-node-k6lkl liveness-probe I1012 10:17:54.811549       1 connection.go:153] Connecting to unix:///csi/csi.sock

**What you expected to happen?**
Don't see this log, at least in log level 1

**How to reproduce it (as minimally and precisely as possible)?**
Just deploy and see the logs

**Anything else we need to know?**:

**Environment**
- Kubernetes version (use `kubectl version`):
- Driver version:
v2.2.0-eks-1-18-2

**Comment by tanvp112** [comment]
For a sizeable cluster, such rapid logging can eat up A LOT of diskspace.

**Comment by keifgwinn** [comment]
please address this, it's such an easy fix

**Comment by janaurka** [comment]
BTW: this can be fixed with overwriting the default values file with:

```
        sidecars:
          livenessProbe:
            image:
              tag: v2.6.0-eks-1-21-13
```
so https://github.com/kubernetes-sigs/aws-efs-csi-driver/pull/595 is essentially too complicated, since it’s also changing the Container Registry.

If the above mentioned solution does not work for you _and_ you are using a k8s version <1.12, then look at the compatible images under: https://gallery.ecr.aws/eks-distro/kubernetes-csi/livenessprobe

If I find the time, I’ll create a PR, to update the `livenessProbe` Image to v2.6.

**Comment by ilvez** [comment]
When driver was installed with Helm, I used this command to update image:
```
kubectl -nkube-system set image deployment/efs-csi-controller liveness-probe=public.ecr.aws/eks-distro/kubernetes-csi/livenessprobe:v2.6.0-eks-1-18-17
```

Your setup may vary -- namespace, deployment name.

**Comment by ilvez** [comment]
The issue is still in the latest chart. I managed to fix it with downloading the chart and replaced sidecars.livenessProbe.image.tag
```
sidecars:
  livenessProbe:
    image:
      repository: public.ecr.aws/eks-distro/kubernetes-csi/livenessprobe
      tag: v2.6.0-eks-1-18-17 # v2.2.0-eks-1-18-13
```
Then applied the upgrade:
```
helm upgrade -f values.yaml --namespace kube-system --set controller.serviceAccount.create=false --set controller.serviceAccount.name=efs-csi-controller-sa aws-efs-csi-driver ../aws-efs-csi-driver-2.2.7.tgz
```
Probably the parameter can be overwritten by invoking helm instead of `values.yaml` change, but I don't use Helm daily, so I didn't bother to look it up. 

Please fix the helm chart :)

**Comment by rajudevop** [comment]
issue: connection.go:153] Connecting to unix:///csi/csi.sock
Can someone explain me what is the root cause of this noisy logs(1 connection.go:153] Connecting to unix:///csi/csi.sock)? Does these relate to errors or cause errors?  How did  this image version (v2.6.0-eks-1-18-17) solved the problem of logs flooded? Please tell me if anyone knows. Thanks :)

**Comment by keifgwinn** [comment]
> issue: connection.go:153] Connecting to unix:///csi/csi.sock Can someone explain me what is the root cause of this noisy logs(1 connection.go:153] Connecting to unix:///csi/csi.sock)? Does these relate to errors or cause errors? How did this image version (v2.6.0-eks-1-18-17) solved the problem of logs flooded? Please tell me if anyone knows. Thanks :)

https://github.com/kubernetes-sigs/aws-efs-csi-driver/issues/564#issuecomment-949517622 has the link to the issue on the liveliness probe, basically noisy logging decision in a sub component (golang connection library).

**Comment by tavin** [comment]
this is fixed in v1.4.9?

**Comment by mskanth972** [comment]
Hi @tavin, yes this is fixed in v1.4.9.
